lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -1763,6 +1812,75 @@ selectSeqStatePendingIxs wid =
   where
     fromRes = fmap (W.Index . seqStatePendingIxIndex . entityVal)
 
+insertScriptPool
+    :: forall k . WalletKey k
+    => W.WalletId
+    -> W.SlotNo
+    -> Seq.VerificationKeyPool k
+    -> SqlPersistT IO ()
+insertScriptPool wid sl pool = do
+    void $ dbChunked insertMany_
+        [ SeqStateAddress wid sl (W.Address payload) (getIndex ix) W.MultisigScript state
+        | (KeyHash payload, (ix, state) )
+        <- Map.toList (Seq.verPoolIndexedKeys pool)
+        ]
+    void $ dbChunked insertMany_ $
+        concatMap toDB $ Map.toList (Seq.verPoolKnownScripts pool)
+  where
+    toAddress = W.Address . xpubToBytes . getRawKey
```

Comment:
! 

That's terribly confusing and could lead to hard-to-find errors. Yet, storing _addresses_ in this table is already a little unhandy, but it's now becoming an impediment since we have two different kind of pools storing different kind of entities. From there I see two options: 

a) Use a completely new table for storing elements of a ScriptPool 
b) Revise the `SeqStateAddress` table to not store serialized addresses, but instead, relax a bit the representation to store mere "payload" that varies depending on the associated role. For payment addresses, it'll be the actual fingerprint (and so we should even gain some performance improvement because we wouldn't have to convert back and forth to and from fingerprints anymore). 

Although it's a bit more complex and will require a _massive_ migration, I am more inclined towards the second option, because it leads to simpler long-term design and also prepare the ground for the upcoming multi-delegation changes. Doing it inside this PR is maybe ill-advised; I can open a side PR to do just that and from which this current PR can branch?

Reply:
Frankly speaking I do not see why adding new table side by side to SeqstateAddress will cause migration problems. indexedKeys in scriptPool is new thing, and new table should not pose problem at first glance (???). Generally, I agree that I re-used the SeqStateAddress table (and expected your remarks here). But it was tempting, only Address was "bent" here.. But it is just Bytestring wrapped, so I thought in the first step it could be some solution (agree it is not the best one)

Reply:
applied in https://github.com/input-output-hk/cardano-wallet/pull/2308/commits/85e56ef477bf5a1fbe02d9208cf5b0aa5ea4cdf9

Reply:
also used indices in knowScripts rather than keys -> https://github.com/input-output-hk/cardano-wallet/pull/2308/commits/73c4cb4aa9a41beb6e953624dc1cc51a399952e5